src/v8_exec.cpp

```diff
@@ -0,0 +1,60 @@
+#include "v8_callbacks.h"
+#include "v8_exec.h"
+
+V8Executor::ExecException::ExecException(std::string exec_exception_str)
+        : exec_exception_str{std::move(exec_exception_str)} {}
+
+const char* V8Executor::ExecException::what() const noexcept {
+    return this->exec_exception_str.c_str();
+}
+
+V8Executor::V8Executor(std::string_view program_name)
+        : isolate{nullptr}, isolate_allocator_ptr{nullptr} {
+    this->init_v8(program_name);
+    this->set_new_isolate();
+    this->set_new_context();
+}
+
+V8Executor::~V8Executor() {
+    if (!this->context.IsEmpty()) {
+        this->context.Reset();
+    }
+
+    if (this->isolate) {
+        this->isolate->Dispose();
+        delete isolate_allocator_ptr;
```

Comment:
Is it ever possible for `isolate_allocator_ptr` to be `nullptr` at this stage?

Reply:
It's legal to delete nullptr

Reply:
:+1: